lang/java/reef-runtime-spark/src/test/java/org/apache/reef/runtime/spark/job/ReefOnSpark.java

```diff
@@ -0,0 +1,85 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.spark.job;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.reef.client.DriverConfiguration;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.proto.ReefServiceProtos;
+import org.apache.reef.client.DriverLauncher;
+import org.apache.reef.runtime.common.REEFEnvironment;
+import org.apache.reef.runtime.yarn.client.unmanaged.UnmanagedAmYarnClientConfiguration;
+import org.apache.reef.runtime.yarn.client.unmanaged.UnmanagedAmYarnDriverConfiguration;
+import org.apache.reef.util.EnvironmentUtils;
+import org.apache.spark.SparkConf;
+import org.apache.reef.tang.Configuration;
+import org.apache.spark.SparkContext;
+
+// Run:
+// ..\spark\bin\spark-submit.cmd
+//     --master yarn --deploy-mode cluster
+//     --class org.apache.reef.examples.hellospark.ReefOnSpark
+//     .\target\reef-examples-spark-0.16.0-SNAPSHOT-shaded.jar
+
+
+/**
+ * Abstraction around submitting a reef job on spark.
+ */
+public final class ReefOnSpark {
+
+  private Logger logger = Logger.getLogger(ReefOnSpark.class.getName());
+
+  private static final String ROOTFOLDER = ".";
+
+  private static final Configuration RUNTIME_CONFIG = UnmanagedAmYarnClientConfiguration.CONF
+      .set(UnmanagedAmYarnClientConfiguration.ROOT_FOLDER, ROOTFOLDER)
+      .build();
+
+  public void process(final String[] args) throws InjectionException {
+
+    logger.setLevel(Level.FINEST);
+
+    SparkConf conf = new SparkConf().setAppName("ReefOnSpark:host");
+    SparkContext sc = new SparkContext(conf);
+
+    try (final DriverLauncher client = DriverLauncher.getLauncher(RUNTIME_CONFIG)) {
+      String jarPath = EnvironmentUtils.getClassLocation(ReefOnSpark.class);
+      Configuration driverConfig = DriverConfiguration.CONF
+          .set(DriverConfiguration.DRIVER_IDENTIFIER, "ReefOnSpark:hello")
+          .set(DriverConfiguration.GLOBAL_LIBRARIES, jarPath)
+          .set(DriverConfiguration.ON_DRIVER_STARTED, ReefOnSparkDriver.StartHandler.class)
+          .set(DriverConfiguration.ON_EVALUATOR_ALLOCATED, ReefOnSparkDriver.EvaluatorAllocatedHandler.class)
+          .build();
+      final String appId = client.submit(driverConfig, 120000);
+      //LOG.log(Level.INFO, "Job submitted: {0} to {1}", Array[AnyRef](appId, jarPath));
+      final Configuration yarnAmConfig = UnmanagedAmYarnDriverConfiguration.CONF
```

Comment:
Same here - we don't need YARN-specific driver for Spark runtime (unless we are in dual-runtime configuration when we need to communicate with YARN directly). In the case of `reef-runtime-spark`, the runtime config is pretty much empty and is just a flag that we are running our REEF app on Spark; for REEF Driver config we should have some class like `SparkDriverConfiguration` (that probably won't be even exposed to the REEF user).
The entire `try` block should be gone, too - I expect that the user-facing code will look more like our `HelloREEF.main()` function.